### PR TITLE
feat: Text Widget Reskinning

### LINKED ISSUE #11000: [Enhancement]: Text widget reskinning
https://www.notion.so/appsmith/Text-da1dc5ca6d7e408fa73734d4acfedd03

### PR DESCRIPTION
**Pull request by dhruvikn**
## Description
This PR fixes UI inconsistencies in the Text widget. Mainly, the truncate button now inherits its color from the theme.

Another PR, #17299, has been raised which adds the changes required to inherit the color from system themes. #17299 should be merged before this PR to enable testing on DPs. 

Fixes #11000 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by dhruvikn** [comment]
> @dhruvikn please confirm this .Currently , I dont see the truncate button color being inherited from theme. Please find loom [here](https://www.loom.com/share/228fff91e4eb4cb69c4b7dba327a33ef)

@kamakshibhat-appsmith 

So the thing with theming is that they require changes on both BE and FE. So I have raised two PRs. [#17298](https://github.com/appsmithorg/appsmith/pull/17298) (FE) and [#17299](https://github.com/appsmithorg/appsmith/pull/17299) (BE).

Firstly, we'll need to merge the backend change [#17299](https://github.com/appsmithorg/appsmith/pull/17299) to release. Once that's done, after some time, the BE changes will start reflecting on [#17298](https://github.com/appsmithorg/appsmith/pull/17298) (FE) DPs.

Then only, we'll be able to see and test the changes. I have also documented this on both the PRs. PFA.

image

image

File: app/client/src/widgets/TextWidget/component/index.tsx [hunk 1]
```diff
@@ -189,6 +189,7 @@ const Content = styled.div<{
     fontSizeUtility(fontSize) || DEFAULT_FONT_SIZE};
 `;
 export interface TextComponentProps extends ComponentProps {
+  accentColor: string;
```
[the remark this answers is not in the document]
Comment (dhruvikn) [reply app/client/src/widgets/TextWidget/component/index.tsx:192]: It's a fallback like this here https://github.com/appsmithorg/appsmith/pull/16910.
Ref - https://theappsmith.slack.com/archives/C02JV8G1MP0/p1663661524522019

**Comment by dhruvikn** [comment]
Can we pick this up in the second half today? Merged https://github.com/appsmithorg/appsmith/pull/17299 sometime ago only. Will take some time to reflect the changes on DP.

cc: @laveena-en @chandannkumar

**Comment by chandannkumar** [comment]
> Can we pick this up in the second half today? Merged #17299 sometime ago only. Will take some time to reflect the changes on DP.
> 
> cc: @laveena-en @chandannkumar

Sure @dhruvikn

**Review by jsartisan — APPROVED** [review]

**Comment by laveena-en** [comment]
@dhruvikn Moving this to dev in progress for now on the basis of our conversation for more insight/clarity.
- Whether the truncate option should inherit the theme's primary colour as soon as it's been dropped onto the canvas.
- If the `deviated from theme` arrow under truncate colour is pertinent to this PR.

**Comment by dilippitchika** [comment]
> Whether the truncate option should inherit the theme's primary colour as soon as it's been dropped onto the canvas.

It should inherit

> If the deviated from theme arrow under truncate colour is pertinent to this PR.

This should also be there

**Comment by dhruvikn** [comment]
Got off a call with @kamakshibhat-appsmith, and she has raised one issue. Toggling to JS and toggling back again removes the inherited color. Fixing this now.

cc: @laveena-en 

https://user-images.githubusercontent.com/22471214/195834080-e921bc61-c5fd-4a4a-8a6f-9a0e2501fc37.mp4

**Comment by kamakshibhat-appsmith** [comment]
Some more inconsistencies regarding truncate button property field
1. on clicking the JS toggle, you can see the binding {{appsmith.theme.colors.primaryColor}}, one clicking the toggle again, you will notice the primary color value is not being retained and the truncate icon is moving to the fall back color(which is black)
2. The tooltip for the deviation icon is moving out of the screen frame.
Screen Shot 2022-10-14 at 4 44 51 PM
@dhruvikn

**Comment by dhruvikn** [comment]
> The tooltip for the deviation icon is moving out of the screen frame.

Can we raise another issue for this? @dilippitchika

**Comment by dhruvikn** [comment]
> on clicking the JS toggle, you can see the binding {{appsmith.theme.colors.primaryColor}}, one clicking the toggle again, you will notice the primary color value is not being retained and the truncate icon is moving to the fall back color(which is black)

Issue has been fixed now.

cc: @kamakshibhat-appsmith @laveena-en

**Review by jsartisan — APPROVED** [review]

**Comment by laveena-en** [comment]
Tested and verified.
P.S @dhruvikn assuming a different PR is to be raised for the tooltip change?

**Comment by dhruvikn** [comment]
> P.S @dhruvikn assuming a different PR is to be raised for the tooltip change?

@laveena-en, waiting for @dilippitchika's thoughts on this here - https://github.com/appsmithorg/appsmith/pull/17298#issuecomment-1278869509

**Comment by dilippitchika** [comment]
Create a new issue for the 2nd one and add this to good first issue. @dhruvikn @laveena-en

**Comment by dhruvikn** [comment]
FYI, here's the issue for the tooltip - https://github.com/appsmithorg/appsmith/issues/17604

cc: @dilippitchika @kamakshibhat-appsmith @laveena-en

**Pull request merged**